### PR TITLE
refined mainImage section to account for responsiveness fixes #842

### LINKED ISSUE #842: Refine UI of the main-image section in the PublicLab Editor
Hi, @TildaDares 
I'm trying to refine the UI of the main-image section for mobile view.
Here's what it currently looks like:

Screenshot 2022-02-16 at 11 00 40 AM\

Here's what I'm trying to achieve:

Screenshot 2022-02-16 at 11 15 41 AM

### PR DESCRIPTION
**Pull request by NARUDESIGNS**
Hello @TildaDares 
This is the PR I've made to address issue #842
So I only made changes to the `PublicLab.Editor.css` file and so I had to use combinations of classes and IDs to target the selected classes. Do have a look and let me know what you think.
Here's what it looks like now:

Screenshot 2022-02-22 at 1 17 53 PM

### CODE REVIEW

**Comment by NARUDESIGNS** [comment]
This is how it was before I started working on it. Also long before now, the uploaded doesn't work for me in my local environment.
I wonder what is wrong but I'd have a look, thanks.

**Comment by TildaDares** [comment]
It'll be better for you to also test how it looks on the plots2 repo. If you have the plots2 repo installed locally, you can do this:
https://github.com/publiclab/PublicLab.Editor#creating-a-mock-server

**Comment by TildaDares** [comment]
Also, remember that plots2 has its own custom styles so you might need to change that too
https://github.com/publiclab/plots2/blob/main/app/views/editor/rich.html.erb#L44-L58

**Comment by NARUDESIGNS** [comment]
> It'll be better for you to also test how it looks on the plots2 repo. If you have the plots2 repo installed locally, you can do this: https://github.com/publiclab/PublicLab.Editor#creating-a-mock-server

I do not have plots2 installed locally. I only use it on Gitpod. But to get you, are you saying we can not make UI changes in the editor project without also making the same change in the plots2 project?

**Comment by TildaDares** [comment]
No, I'm saying to know that the UI looks good, you need to test image upload which is why you need the plots2 repo. Also, the plots2 repo has it's own custom styling, so you need to also keep that in mind while you're adding styles.

**Comment by jywarren** [comment]
Hi all, thanks for this, the refinement proposed in https://github.com/publiclab/PublicLab.Editor/issues/842 looks great. 

@NARUDESIGNS basically this is the situation -- the HTML actually displayed on PublicLab.org is from plots2. But since we develop the editor in its own library, which doesn't contain plots2 code, we create an "example" HTML file, just so we can properly run the code, test it out, etc. It's called an example also because if anyone else ever uses this editor, they know what HTML is needed for the library to work properly. 

But that means that if you want to style how something looks differently, you need to do it on plots2, which is what's actually published to the site. That also means that we need to either:

1. not make any changes which would break the version still at https://github.com/publiclab/PublicLab.Editor/blob/main/examples/index.html, or
2. update https://github.com/publiclab/PublicLab.Editor/blob/main/examples/index.html so that the example still works

option 2 above is technically a breaking change. That's OK, but if we do this, perhaps we can do it all at one time, rather than in a series of PRs, since we would technically need to make a major version bump each time. 

That's a good reason to make as much of our changes as we can without requiring different underlying HTML. We could do all our changes to the HTML in one PR even before we do all our styling fixes, so that we only need to bump the major version once. That might be complex to think through, and if you need to make 2 breaking changes, it's not such a big deal. But if you think it may be relatively easy to do all your HTML improvements in a single big change, or to even avoid breaking HTML changes entirely, that's ideal. Many HTML changes will result in a different appearance, but the original example file at https://github.com/publiclab/PublicLab.Editor/blob/main/examples/index.html will still work, right?

I hope this makes sense. Some HTML is done actively using JavaScript, rather than relying on existing HTML. But the template file at rich.html contains essentially everything which /must/ be installed already in the final application for the Editor to have a foundation to install upon. Make sense? I'm happy to discuss more!

**Comment by NARUDESIGNS** [comment]
OK I think I'd have to go through again to grasp it well but these are where my confusion lies:
1. I only made changes to `PublicLab.Editor.css` not the HTML file.
2. Other CSS UI changes we've done in the past weren't complicated as this seems. We merged them and they worked just fine. 

Can you speak to these concerns? @jywarren

**Comment by NARUDESIGNS** [comment]
I had the same old problem we had with plots2 on Gitpod last time 😭 😭 😭 

Screenshot 2022-03-07 at 4 57 41 PM

**Comment by NARUDESIGNS** [comment]
> copy the folder /public/lib/publiclab-editor/dist/ to /public/lib/publiclab-editor/dist-bkp/

@jywarren @TildaDares can I try something like this with the plots2 project locally?

**Comment by jywarren** [comment]
For reference here was the last time this happened: https://github.com/publiclab/plots2/pull/10711

**Comment by NARUDESIGNS** [comment]
Alright, I'd try that, thanks.
@jywarren

**Comment by NARUDESIGNS** [comment]
Thanks for the tip @jywarren that was really helpful.
@TildaDares can you please have a look again and see if things show up better now?

**Comment by TildaDares** [comment]
Great work @NARUDESIGNS! This is what it looks like now
Screenshot 2022-03-10 at 12 21 39

At about 767px width, the choose file button kind of cuts off from the screen. This happens till around 550px.

https://user-images.githubusercontent.com/63427719/157653082-6f617317-a85d-4540-85b2-08218c2acdd7.mov

**Comment by NARUDESIGNS** [comment]
Oh, I see, I guess we need to target screens within that range as well. I think that should be tablet view right?
@TildaDares

**Comment by TildaDares** [comment]
@NARUDESIGNS yeah you’re right

**Comment by NARUDESIGNS** [comment]
@TildaDares I think we should be fine now. Please check if all looks good so we can merge!
Thank you!!

File: dist/PublicLab.Editor.css [hunk 1]
```diff
@@ -320,6 +320,53 @@ https://github.com/sliptree/bootstrap-tokenfield/pull/287 */
   background-color: #007bff !important;
 }
 
+@media (max-width: 800px) {
```
Comment (TildaDares) [inline dist/PublicLab.Editor.css:323]: Change this to about `992px` and we should be good!

Comment (NARUDESIGNS) [reply dist/PublicLab.Editor.css:323]: Alright then
On it right away

Comment (NARUDESIGNS) [reply dist/PublicLab.Editor.css:323]: But why exactly 992 if I may ask?

Comment (NARUDESIGNS) [reply dist/PublicLab.Editor.css:323]: Ok then, looks great on my end but lets try.
Also can you confirm it looks great and 993px and wider on your end?

Comment (NARUDESIGNS) [reply dist/PublicLab.Editor.css]: Awesome!

**Comment by NARUDESIGNS** [comment]
Thank you so much for your time @TildaDares I think we can merge now
cc @jywarren

File: dist/PublicLab.Editor.css (hunk 1 above)
Comment (jywarren) [reply dist/PublicLab.Editor.css]: Haha love this quick iteration, you two! Thanks! 🎉

**Comment by NARUDESIGNS** [comment]
Thank you very much!
I'd move on to another part of the editor and let you know what I'm working on
@TildaDares @jywarren

**Pull request merged**